File: src/main/java/org/kite9/diagram/position/VPos.java
/**
 * 
 */
package org.kite9.diagram.position;

public enum VPos {
	
	
	UP, DOWN;
	
	
	
	public Direction getDirection() {
		if (this.ordinal()==0) {
			return Direction.UP;
		} else {
			return Direction.DOWN;
		}
	}
	
	public static VPos getFromDirection(Direction d) {
		switch (d) {
		case UP:
			return UP;
		case DOWN:
			return DOWN;
		default:
			return null;
		}
	}
}

File: visualization/kite9-visualization-common/src/jvmMain/kotlin/org/kite9/diagram/model/position/VPos.kt
/**
 *
 */
package org.kite9.diagram.model.position

import org.kite9.diagram.model.position.VPos

enum class VPos {
    UP, DOWN;

    val direction: Direction
        get() = if (ordinal == 0) {
            Direction.UP
        } else {
            Direction.DOWN
        }

    companion object {
        fun getFromDirection(d: Direction?): VPos? {
            return when (d) {
                Direction.UP -> UP
                Direction.DOWN -> DOWN
                else -> null
            }
        }
    }
}